internal/powerbi/resource_pbix.go

```diff
@@ -237,14 +259,14 @@ func deletePBIX(d *schema.ResourceData, meta interface{}) error {
 
 	groupID := d.Get("workspace_id").(string)
 
-	if reportID := d.Get("report_id"); reportID != nil && reportID != "" {
+	if reportID, reportIDOk := d.GetOk("report_id"); reportIDOk {
 		err := client.DeleteReportInGroup(groupID, reportID.(string))
 		if err != nil {
 			return err
 		}
 	}
 
-	if datasetID := d.Get("dataset_id"); datasetID != nil && datasetID != "" {
+	if datasetID, datasetIDOk := d.GetOk("dataset_id"); datasetIDOk {
```

Comment:
As commented in the other PR, wouldn't it be required to add a condition `datasetID != rebindDatasetId` here? Otherwise the dataset is deleted here, even though it might be managed by another resource.

Reply:
Nevermind, I see how you solved this by only importing the reportId and datasetId on initial creation of the resource.

I have tested your version now with our datasets and reports and all seems to be working fine :) Remove operation also worked smoothly for me.